### PR TITLE
Allow passing extra arguments to eval release command

### PR DESCRIPTION
**Pull request by hauleth**
It is handy for writing custom actions to communicate and operate with the release.

### CODE REVIEW

**Comment by hauleth** [comment]
At first I added it to `mix eval` as well, but I wasn't sure it will work. First I need to make it work on Windows, but I know nothing about Batch so I am testing how I can make it work.

**Comment by josevalim** [comment]
@hauleth I believe you can change System.argv on the fly. Or perhaps see how `mix run` does it.

**Comment by wojtekmach** [comment]
@josevalim I don't think we ever tackled `bin/app eval`. We were mostly concerned about `bin/app start` (https://github.com/elixir-lang/elixir/pull/11426#issuecomment-1041422339) and a new `bin/app rpc MOD FUN ...ARGS` (https://github.com/elixir-lang/elixir/pull/11661)

**Comment by hauleth** [comment]
Ehhhh… Do we need to support Batch? Cannot we support PowerShell on Windows instead?

https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/shift

> The shift command has no effect on the `%*` batch parameter.

Ok, so looping, here we go.

**Comment by wojtekmach** [comment]
We can’t default to .ps1 as executing these is disabled by default on Windows.

**Comment by hauleth** [comment]
@josevalim `eval` is done, the `rpc` is not, but honestly I do not know if that is something we should do. The thing is that `rpc` is called in context of the application, so that mean that we would not be able to do `rpc "IO.inspect(System.argv())"` to get the args that were passed to `start` command. I think that `eval` is the only one where it is sensible to have access to passed args. For `rpc` there would be another mechanism needed for that.

**Comment by josevalim** [comment]
Yup. We cannot do `rpc`.

**Comment by hauleth** [comment]
So now it is ready, works on Windows (at least tests says so) and on *nixes.

**Comment by josevalim** [comment]
Thank you @hauleth! We just need to support the same on the new `mix eval` task and we are good!

**Comment by hauleth** [comment]
There is new command for `mix` beyond `mix run`? Ok, I will take a look.

**Comment by josevalim** [comment]
`mix eval` was designed to mirror the VM state from `bin/app eval` to make it easier for testing and reuse!

**Comment by hauleth** [comment]
Added the support for arguments in `mix eval`

File: lib/mix/test/mix/tasks/eval_test.exs [hunk 1]
```diff
@@ -26,6 +26,30 @@ defmodule Mix.Tasks.EvalTest do
     end)
   end
 
+  test "can accept arguments", context do
```
Comment (josevalim) [inline lib/mix/test/mix/tasks/eval_test.exs:29]:
```suggestion
  test "accepts arguments", context do
```

File: lib/mix/test/mix/tasks/eval_test.exs [hunk 2]
```diff
@@ -26,6 +26,30 @@ defmodule Mix.Tasks.EvalTest do
...
+    in_tmp(context.test, fn ->
+      Mix.Tasks.Eval.run(["send self(), {:argv, System.argv()}", "bar", "baz"])
+      assert_received {:argv, ~w[bar baz]}
+    end)
+  end
+
+  test "reset argv to previous value", context do
```
Comment (josevalim) [inline lib/mix/test/mix/tasks/eval_test.exs:36]:
```suggestion
  test "resets argv to previous value", context do
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**